openedx/adg/lms/applications/helpers.py

```diff
@@ -330,3 +332,47 @@ def is_user_qualified_for_bu_prereq_courses(user):
         hasattr(user, 'application') and
         user.application.business_line
     )
+
+
+def get_users_with_active_enrollments_from_course_groups(user_ids, course_groups):
+    """
+    Given the ids and course groups, filter users enrolled in the course group courses.
+
+    Args:
+        user_ids (list): List of user ids
+        course_groups (list): List of MultilingualCourseGroup objects
+
+    Returns:
+        users_with_active_enrollment (list): list of user_ids who has active enrollment from the course groups
+    """
+    users_ids_with_active_enrollment = []
+
+    for course_group in course_groups:
+        course_keys = course_group.open_multilingual_course_keys()
+        users_ids_with_active_enrollment += User.objects.filter(
+            id__in=user_ids,
+            courseenrollment__course__in=course_keys,
+            courseenrollment__is_active=True,
+        ).values_list('id', flat=True).distinct()
```

Comment:
We can simply return users here instead of extracting user ids as a result of which we won't have to filter users based on ids in `check_users_for_application_update` method in both commands